 <div style="display:inline-block; position:absolute; top:0; right:0; color: #230e3e; font:italic 1em serif;">$\LaTeX$<span style="display:inline-block; vertical-align:bottom">
    <a href="https://www.mathjax.org">
    <img style="height:1.5em" title="Powered by MathJax"
    src="https://www.mathjax.org/badge/mj_logo.png"
    border="0" alt="Powered by MathJax" >
</a></span></div>

<div style="display:inline-block; position:absolute; top:0; left:0; color: #230e3e; font:italic 1em serif;"><a href="https://www.julialang.org">
    <img style="height:2.25em; vertical-align:top" title="Powered by Julia"
    src="./figures/julia-logo-mask.png"
    border="0" alt="Powered by Julia" >
</a></div>
</a></div>$\require{cancel} \require{mhchem} \require{bbox} \require{color} \require{enclose} \require{AMSmath} \require{AMSsymbols}$

**Water vapor enters a subsonic diffuser at 0.5 bar, 175 °C, and 200 m/s. The diffuser inlet is 100 cm$^2$. During passage through the diffuser, the fluid velocity is reduced to 50 m/s, pressure increases to 1.0 bar and heat transfer to the surroundings is 0.5 kg/kJ. Determine (1) the final temperature, (2) the mass flow rate, in kg/s and (3) the outlet area, in cm².** 

### Given
<ol> 
 $$\ce{H2O} \left| 
 \begin{array}{ll} 
   P_{1} = 0.5\, \ce{bar}       \\[0pt]
   T_{1} = 175\, \ce{°C}         \\[0pt]
   \vec V_{1} = 200\, \ce{m/s}   \\[0pt]
   A_{1} = 100\, \ce{cm^2}       \\[0pt]
 \end{array}\right|
 \xrightarrow{\normalsize q_{out} = 0.5\, \ce{kJ/kg}}
 \left| 
 \begin{array}{ll} 
   P_{2} = 1\, \ce{bar}       \\[0pt]
                              \\[0pt]
   \vec V_{2} = 50\, \ce{m/s} \\[0pt]
                              \\[0pt]
 \end{array}\right|
 $$


In [1]:
using Unitful       # Unitful units package

In [2]:
# Given 
P₁ = 0.5u"bar"  
T₁ = 175u"°C"
V⃑₁ = 200u"m/s"
A₁ = 100u"cm^2"

P₂ = 1u"bar"  
V⃑₂ = 50u"m/s"
q_out = 0.5u"kJ/kg";

### Find 
1. $T_2$  
2. $\dot m$  
3. $A_2$

### Get Organized  and Attack
- This time, the conservation of mass is trivial, but we must calculate the factors that determine mass flow. 
- We'll attack first using mass conservation, and follow up with energy conservation to account for the heat loss. 
- We also note that the diffuser is a steady-state device.
- We will account for changes in kinetic energy, assume neglible potential energy effects.
- We need to resolve State 2. We have pressure, so we another property, independednt of P. 

#### Mass Conservation

\begin{align*}
    \dot m_{in} - \dot m_{out} &= \frac{d}{dt} m_{sys} \\[6pt]
    \dot m_1 - \dot m_2 &= 0 \\[12pt]
\end{align*}   
$$$$
\begin{align*}
    \dot m_1 &= \dot m_2 \\[6pt]
    \frac{\vec V_1 A_1}{v_1} &= \frac{\vec V_2 A_2}{v_2}
\end{align*}  

#### Energy Conservation

\begin{array}{cccl}
  \dot E_{in} &-& \dot E_{out} &=& \frac{d}{dt} E_{sys} \\[6pt]
  \displaystyle \dot m \left(h_1 + \frac{\vec V_1^2}{2} \right)  &-& \displaystyle \dot m
     \left( h_2 + \frac{\vec V_2^2}{2} \right) - \dot m q_{out} &=& 0 \\[12pt]
  \displaystyle h_1 + \frac{\vec V_1^2}{2} &-& \displaystyle h_2 - \frac{\vec V_2^2}{2} 
  - q_{out} &=& 0 \\[12pt]
\end{array}  

These equations can get out of hand, so now is a good time to stop and see what we know and don't know.

$$ \color{blue} h_1 \color{black} + \frac{\color{green}\vec V_1^2}{2} -  \color{red}h_2 \color{black} - \frac{\color{green}\vec V_2^2}{2} 
  - \color{green}q_{out}\color{black} =0 $$
  
We can get $h_1$ from the <b style="color:blue">TABLES</b>;  $\vec V_1, \vec V_1,$ and  $q_{out}$ are <b style="color:green">GIVEN</b>; and $h_2$ is our only <b style="color:red">UNKNOWN</b>. We're in great shape: $h_2$ gives us that second property we needed to set State 2.

$$\ce{H2O} \left| 
 \begin{array}{ll} 
   P_{1} = 0.5\, \ce{bar}       \\[0pt]
   T_{1} = 175\, \ce{°C}         \\[0pt]
 \end{array}\right| \longrightarrow 
h_1 = \textit{double interpolation!} \quad 
 \begin{align*}
   0.35\, \ce{bar} < P_{1}& < 0.70\, \ce{bar}       \\[0pt]
   160\, \ce{°C} < T_1& <  200\, \ce{°C}    \\[0pt]
 \end{align*}
$$ 


Stay organized, and you'll get through it unscathed. Interpolate in one property, then the other. You can interpolate both simultaneously, but it's very easy to miss a little mistake. 

<div class="panel-warning"><div class="panel-heading">$$\small \textit{I left the units off the weighting fraction to reduce clutter a bit.}$$</div></div>

In [3]:
h_35kPa_160°C = 2800.6u"kJ/kg"
h_35kPa_200°C = 2878.4u"kJ/kg"

h_35kPa_175°C = h_35kPa_160°C + (h_35kPa_200°C - h_35kPa_160°C) * (175 - 160)/(200 - 160);
# println("h_35kPa_175°C = ",h_35kPa_175°C)

In [4]:
h_70kPa_160°C = 2798.2u"kJ/kg"
h_70kPa_200°C = 2876.7u"kJ/kg"

h_70kPa_175°C = h_70kPa_160°C + (h_70kPa_200°C - h_35kPa_160°C) * (175 - 160)/(200 - 160);
# println("h_70kPa_175°C = ",h_70kPa_175°C)

In [5]:
h_50kPa_175°C = h_35kPa_175°C + (h_70kPa_175°C - h_35kPa_175°C) * (50 - 35)/(70 - 35)

h₁ = h_50kPa_175°C;
println("h₁ = ",h₁)

h₁ = 2828.473214285714 kJ kg^-1


<p style = " background-image:  url(./figures/bacgroundTiles.png); line-height: 0.35em" >&nbsp;</p>

#### Resolve State 2
We seek $T_2, \dot m,$ and $A_2.$ We can get them all from State 2. 

In [6]:
h₂ = h₁ +  (V⃑₁^2 - V⃑₂^2) / 2 - q_out |>u"kJ/kg"
println("h₂ = ",h₂)

h₂ = 2846.723214285714 kJ kg^-1


$\ce{H2O} \left| 
 \begin{array}{ll} 
   P_{2} = 1\, \ce{bar}       \\[0pt]
   h_{2} = 2846.72\, \ce{kJ/kg}        
 \end{array}\right|
$

#### Get $T_2$

In [7]:
h_1bar_160°C = 2796.2u"kJ/kg"
h_1bar_200°C = 2875.3u"kJ/kg"

T₂ = 160u"°C" + (200 - 160)u"K" * (h₂ - h_1bar_160°C) / (h_1bar_200°C - h_1bar_160°C) |> u"°C"
println("T₂ = ",T₂)

T₂ = 185.54903377280112 °C


##### <div class="panel-success"><div class="panel-heading"><div class="badge"></div> $$\ \normalsize T_{2} = 185.55\, \ce{°C} $$
</div></div>

<p style = " background-image:  url(./figures/bacgroundTiles.png); line-height: 0.35em" >&nbsp;</p>

#### Calculate $\dot m$
We can calculate $\dot m$ from State 1 or State 2. 

$$ \frac{\vec V_1 A_1}{v_1} = \frac{\vec V_2 A_2}{v_2} $$

We have an unfortunate selection of variables. We need to calculate both $v_1$ and $v_2$ to finish up.

##### Get $v_2$

In [8]:
v_1bar_160°C = 1.984u"m^3/kg"
v_1bar_200°C = 2.172u"m^3/kg"

v₂ = v_1bar_160°C + (v_1bar_200°C - v_1bar_160°C) * (h₂ - h_1bar_160°C) / (h_1bar_200°C - h_1bar_160°C) 
println("v₂ = ",v₂)

v₂ = 2.1040804587321653 m^3 kg^-1


##### Get $v_1$
In case you didn't get enough double interpolation the first time.

In [9]:
v_35kPa_160°C = 5.696u"m^3/kg"
v_35kPa_200°C = 6.228u"m^3/kg"

v_35kPa_175°C = v_35kPa_160°C + (v_35kPa_200°C - v_35kPa_160°C) * (175 - 160)/(200 - 160);
# println("v_35kPa_175°C = ",v_35kPa_175°C)

In [10]:
v_70kPa_160°C = 2.841u"m^3/kg"
v_70kPa_200°C = 3.108u"m^3/kg"

v_70kPa_175°C = v_70kPa_160°C + (v_70kPa_200°C - v_35kPa_160°C) * (175 - 160)/(200 - 160);
# println("v_70kPa_175°C = ",v_70kPa_175°C)

In [11]:
v_50kPa_175°C = v_35kPa_175°C + (v_70kPa_175°C - v_35kPa_175°C) * (50 - 35)/(70 - 35)

v₁ = v_50kPa_175°C;
println("v₁ = ",v₁)

v₁ = 4.1705 m^3 kg^-1


<p style = " background-image:  url(./figures/bacgroundTiles.png); line-height: 0.35em" >&nbsp;</p>

#### Back to $\dot m$

$$ \dot m = \frac{\vec V_1 A_1}{v_1} = \frac{\vec V_2 A_2}{v_2}$$

In [12]:
ṁ = V⃑₁ * A₁ / v₁ |> u"kg/s"

0.47955880589857336 kg s^-1

<div class="panel-success"><div class="panel-heading"><div class="badge"></div> $$\ \normalsize \dot m = 0.48\, \ce{kg/s} $$
</div></div>

In [13]:
A₂ =  ṁ * v₂ / V⃑₂ |> u"cm^2"

201.80606246082394 cm^2

<div class="panel-success"><div class="panel-heading"><div class="badge"></div> $$\ \normalsize A_2 = 201.81\, \ce{cm^2} $$
</div></div>

_...that was painful..._

<footer style="background-color: #461d7c; color:rgba(253,208,35,0.35); line-height:2em; padding-left:1em; padding-right:0.5em; bottom:0; float:bottom; text-align:right;"><i>ψᵟ</i></footer>